packages/titus-backend/lib/plugins/auth/authzcheck.js

```diff
@@ -0,0 +1,58 @@
+'use strict'
+
+const { Forbidden } = require('http-errors')
+const fp = require('fastify-plugin')
+
+async function authzChecks(server, options) {
```

Comment:
we should not apply conditional logic in this route depending on the provider. we have two options:

- if the route has a different implementation depending on the provider, then we should register the route conditionally as is already happening when each of the provider's plugins is registered conditionally
- otherwise we keep the implementation of the route the same but  move the conditional logic elsewhere. in other words, we should not see if(provider === "...") anywhere (or at least not in any new places in the codebase), as it's the wrong approach